Spigot-API-Patches/0005-Adventure.patch

```diff
@@ -413,22 +413,23 @@ index 7c715fdc11ab7837552b1fe3ffd08b31cec0a63b..426b1e83226e674ee4bf3ec05ddcd3ac
      public static int broadcastMessage(@NotNull String message) {
          return server.broadcastMessage(message);
      }
-@@ -836,6 +838,7 @@ public final class Bukkit {
+@@ -836,6 +838,32 @@ public final class Bukkit {
          server.shutdown();
      }
  
 +    // Paper start
-     /**
-      * Broadcasts the specified message to every user with the given
-      * permission name.
-@@ -845,6 +848,21 @@ public final class Bukkit {
-      *     permissibles} must have to receive the broadcast
-      * @return number of message recipients
-      */
-+    public static int broadcast(@NotNull net.kyori.adventure.text.Component message, @NotNull String permission) {
-+        return server.broadcast(message, permission);
++    /**
++     * Broadcast a message to all players.
++     * <p>
++     * This is the same as calling {@link #broadcast(java.lang.String,
++     * java.lang.String)} to {@link Server#BROADCAST_CHANNEL_USERS}
++     *
++     * @param message message to broadcast
++     * @return number of message recipients
++     */
++    public static int broadcast(@NotNull net.kyori.adventure.text.Component message) {
```

Comment:
What about instead implementing `ForwardingAudience`?

Reply:
`Server` already implements `ForwardingAudience`, this is just the addition of a method that was missing providing the ability to broadcast using just a component without having to manually specify the permission for broadcasts.

Essentially, the reason this method was added is because the `BroadcastMessageEvent` exists and currently isn't being fired for `Component` versions of the `broadcast` methods. The options are either saying that `Server#sendMessage` counts as a broadcast, and thus should fire this event, or distinguishing between a "server broadcast" and a "message sent to everyone in this audience", which this PR does. This means that only the `broadcast` methods will fire the `BroadcastMessageEvent`, which I think makes sense.